### ISSUE TITLE
[C] can not change CFLAGS option in Makefile on the fly

### ISSUE THREAD

**Issue by junjiemars** [kanaka/mal#397]
It's not a issue, 
Under the ```mal/c``` directory,
```sh
# I need easy debugging
CFLAGS="-g -O0" make clean step0_repl
```
will generate the following output:
```sh
gcc -g -O0 -g -O2 -DOSX=1 -I/opt/local/include/glib-2.0 -I/opt/local/lib/glib-2.0/include -I/opt/local/include -I/opt/local/lib/libffi-3.2.1/include -c step0_repl.c -o step0_repl.o
```
so ```-O2``` will wins, but I want to let ```CFLAGS``` includes ```-O0``` not ```-O2```.

**Comment by junjiemars** [comment]
It's works
```diff
modified   c/Makefile
@@ -1,6 +1,6 @@
 USE_READLINE ?=
 USE_GC ?= 1
-CFLAGS += -g -O2
+CFLAGS ?= -g -O2
 LDFLAGS += -g
```

**Comment by kanaka** [comment]
This was resolved by #403
